### DISCUSSION TITLE
SDL's kmsdrm mode with multiple connectors: How do I specify which DRM display gets the output?

### DISCUSSION THREAD

**Discussion by phaserblast** [mamedev/discussions#143] [General]
Linux user here. I see in SDL's SDL_kmsdrmvideo.c source that all DRM connectors are enumerated and added to the list of SDL displays. But, how do I tell MAME which output to use? I've tried various combinations of -screen 0,1,2 etc. and MAME always renders to the first connector it finds. Is there a way to force it to use a particular SDL display?

**Comment by galibert** [comment]
At least for me the names of the screens are screen0, screen1, etc.  To
pass to -screen or to put in the screen entry of mame.ini.

  OG.

**Reply by phaserblast** [reply]
Thanks, Olivier. Nice one, what you suggested works to select the correct connector. However, on my laptop, there is still severe tearing and sync problems if the internal display is left running while outputting to the HDMI display using VRR. I use an AMD-based laptop with a 780M GPU. This is fixed if I disable the CRTC using KMSDRM_drmModeSetCrtc() by setting the connectors, count and mode to NULL,0,NULL. It seems the GPU can't figure out correct VRR output when multiple connectors are active.

Perhaps there's a way to select the display in "exclusive" mode. I tried a quick patch to SDL which disables all unused CRTCs and this works, but if there's a way to do it through MAME that would be better, obviously.
